apps/backoffice-v2/src/pages/Entity/components/Case/case-action-variants/WebsiteMonitoringCaseActions/hooks/useWebsiteMonitoringCaseActions/useWebsiteMonitoringCaseActions.ts

```diff
@@ -0,0 +1,70 @@
+import { IUseActions } from '@/pages/Entity/components/Case/hooks/useCaseActionsLogic/interfaces';
+import { useCaseActionsLogic } from '@/pages/Entity/components/Case/hooks/useCaseActionsLogic/useCaseActionsLogic';
+import { StateTag } from '@ballerine/common';
+import { useCallback, useState } from 'react';
+
+export interface IResolutionActionsFormValues {
+  action: string;
+  comment: string;
+}
+
+export const useWebsiteMonitoringCaseActions = ({ workflowId: id, fullName }: IUseActions) => {
+  const {
+    tag,
+    assignedUser,
+    authenticatedUser,
+    isLoadingCase,
+    assignees,
+    isLoading,
+    workflow,
+    isActionButtonDisabled,
+    onMutateRejectEntity,
+    onMutateApproveEntity,
+    onMutateAssignWorkflow,
+  } = useCaseActionsLogic({ workflowId: id, fullName });
+
+  const [formValues, setValues] = useState<IResolutionActionsFormValues>({
+    action: '',
+    comment: '',
+  });
+
+  const handleResolutionActionChange = useCallback(
+    (value: string) => setValues(prev => ({ ...prev, action: value })),
+    [],
+  );
+
+  const handleResolutionCommentChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setValues(prev => ({ ...prev, comment: event.target.value }));
+    },
+    [],
+  );
+
+  const handleSubmit = useCallback(() => {
+    if (formValues.action === 'resolved') {
```

Comment:
Replace the string literals with an enum or make sure the type of `action` is a union.

Reply:
P1